### ISSUE TITLE
[suggestion] moving from TabTree, I'm missing some convenient features

### ISSUE THREAD

**Issue by najodleglejszy** [tabkit/tabkit2#105]
hi! I decided to try Tab Kit, and I really like it. I'm planning to stay with it, since it supports hardware acceleration (while TabTree is having problems with it on certain GPUs), but there's some stuff that I'm missing.

1. the ability to pin some tabs when using vertical tab bar. 

2. scrolling over tab bar to switch tabs - makes switching tabs easier, as I don't have to hunt and peck a tab that I want to move to

3. compact title bar - basically, the hide/maximise/close buttons get moved next to the address bar.

basically, this is the look that I can't fully recreate with Tab Kit.

![firefox_2016-03-28_16-44-28](https://cloud.githubusercontent.com/assets/8598559/14080406/34963f50-f505-11e5-8219-22ea857fd659.png)

**Comment by conquerist** [comment]
2. Have you tried installing the Tab Kit [mouse gestures add-on](https://addons.mozilla.org/en-US/firefox/addon/tab-kit-mouse-gestures)? 

![scrolling](http://i.imgur.com/uS8n9ML.png)

3. The add-on [Hide Caption Titlebar Plus](https://addons.mozilla.org/en-US/firefox/addon/hide-caption-titlebar-plus-sma/) can take of that.

![hide caption titlebar plus](http://i.imgur.com/DnEOnHp.png)

**Comment by PikachuEXE** [comment]
Thank @conquerist for answering some questions :P
I will answer for the rest
1)
It is disabled on purpose since it does not work well with Tab Kit's sorting

**Comment by najodleglejszy** [comment]
I only just noticed that I got some replies to this thread. thanks for the answers! that's the first time I've heard of those two addons, they're working flawlessly. I'm hoping the tab pinning will get sorted out in the future :)

**Comment by PikachuEXE** [comment]
@najodleglejszy 
Can you description what "tab pinning" means?
Is that the same as Firefox's built in "Pin Tab" feature?

**Comment by najodleglejszy** [comment]
yeah, I meant the stock Firefox feature. I can see it kind of works right now though,

**Comment by najodleglejszy** [comment]
yeah, I meant the stock Firefox feature. I can see it kind of works right now though - after pinning the tab can't be placed between non-pinned tabs, as intended. when the tab is pinned, the tab title shifts to the left, and it's looking weird to be honest. here's the comparison:

the first tab is unpinned
![firefox_2016-05-05_14-41-03](https://cloud.githubusercontent.com/assets/8598559/15043746/bd5cdfb4-12cf-11e6-90f4-b87c8ce79f88.png)

the first tab is pinned
![firefox_2016-05-05_14-41-14](https://cloud.githubusercontent.com/assets/8598559/15043755/cd0af90a-12cf-11e6-85a0-92aee54d961e.png)

**Comment by PikachuEXE** [comment]
I don't think the stock behaviour is compatible with TK's tab grouping.
Also a pinned tab will look similar to normal tabs.

Also I think pinned tab is just a clone of Chrome's feature 
(Maybe the FF dev is trying to make Firefox a Chrome like browser?)
And Chrome has that feature since there is **NO** tab grouping in its design
(The only browser with some tab grouping in design is Vivaldi)

I am not sure why that features is required.

**Comment by najodleglejszy** [comment]
actually, Firefox had that feature introduced in v5 and I’m almost 100% sure it was before Chrome had it :) the pinning is useful when you want to have a couple of tabs open constantly in background, but you don’t want to have them take too much place. regarding them looking similar to regular tabs: check the first screenshot I posted, there are 3 pinned tabs in theb’ sidebar when using TabTree.

**Comment by PikachuEXE** [comment]
For me that feature has these purposes:
- Grouping the tabs
- locking the group position
Anything else?

**Comment by jjeziorny** [comment]
The "Pin Tab" feature does not work on vertical mode, tabkit even throws a message saying tabkit doesn't support App Tabs in vertical mode.

**Comment by PikachuEXE** [comment]
Sorry that's intended
I can only maintain the extension, can't really develop any feature 
(and FF breaks stuff too often)

**Comment by PinkCathodeCat** [comment]
I see you're developing again - any chance of getting this to work? I'd do it myself but I'm a bit useless at coding :(

**Comment by PikachuEXE** [comment]
If you mean "Pin Tab" in vertical mode
I can remove the lock, but won't fix issue around it

**Comment by PinkCathodeCat** [comment]
I want a horizontal pin tab list at the top of the vertical tree - is that not possible now? Tree style tabs had it, but it broke in Palemoon 28 so I started using TabKit.
